docs/framework/social-post-generator.mdx

```diff
@@ -2,218 +2,448 @@
 
-    ![Text input settings](/framework/images/tutorial/social_post/sp_gen_8.png)
+  <Step title="Bind the Text component to the 'state' variable's 'posts' key">
+  Select the **Text** component. In the **Properties** panel:
+  
+  - Find the **Text** property, which specifies the content of the **Text** component. Set its value to `@{posts}`.
+  - Set the **Use Markdown** property to **yes**.
   </Step>
-  <Step title="Set visibility for section containing tags and posts">
-    Finally, click on the **Section** that contains the Tags and Text components. Under Visibility, choose **Custom** and set it to `posts`.
+
+  <Step title="Set the visiblity of the Section component containing the Tags and Text components based on the 'state' variable's 'posts' key">
+  Select the **Section** component containing the **Tags** and **Text** components. In the **Properties** panel:
+
+  - Scroll to the **Visibility** property at the bottom.
+  - Click on the **Custom** button.
+  - In the **Visibility value** field, set the value to `posts`. This will cause the **Section** to be visible only when the `state` variable's `posts` key has a non-empty value.
   </Step>
 </Steps>
 
-When you save and run this code, your social media posts will generate in the text section you created earlier.
 
-![Finished social post generator application](/framework/images/tutorial/social_post/sp_gen_9.png)
+## Test the application
+
+You've completed all the steps to make a working social post generator, and you can try using it right now, even while editing the user interface!
+
+Enter a topic into the **Topic for social posts and tags** text field, then click the **Generate Posts** button* _twice_ — the first time will cause the **Properties** panel to appear, and the second click will register as a click. You'll know that you've clicked the button when you see the **Message** component display the text “Generating social media posts and tags for you...”
+
+![The working social post generator, with the project editor in "UI" mode](/framework/images/tutorial/social_post/sp_gen_2e.png)
+
+To get a better sense of what the experience will be like for the user, switch to the preview by changing the edit mode (located near the upper left corner of the page) from _UI_ mode to _Preview_ mode by selecting the **Preview** option:
+
+![The working social post generator, with the project editor in "Preview" mode](/framework/images/tutorial/social_post/sp_gen_2f.png)
+
+You can see the output of any `print()` functions and error messages by clicking on the **Log** button located near the upper right corner of the page:
 
+![The working social post generator, with the project editor in "Preview" mode and the log pane displayed](/framework/images/tutorial/chat/chat_assistant_2g.png)
```

Comment:
@AccordionGuy this image isn't displaying for me, may need to double-check it

Reply:
Fixed!